addon/helpers/href-to.js

```diff
@@ -2,7 +2,17 @@ import Helper from '@ember/component/helper';
 import { getOwner } from '@ember/application';
 
 function hrefTo(context, targetRouteName, ...rest) {
-  let router = getOwner(context).lookup('service:router');
+  let owner = getOwner(context);
+
+  // If used in an engine, namespace the route
+  if(owner.mountPoint && targetRouteName !== 'application') {
```

Comment:
perhaps create a well named variable in place of the comment? Something like `linksToEngine` or `linksToOtherEngine` perhaps?